forcedotcom/sfdx-core
# fix: identify sandboxes during authentication

### What does this PR do?
Enhances `AuthInfo.identifyPossibleScratchOrgs()` to also identify sandboxes.
Adds another public method to `Org` to allow querying the `SandboxProcess` sObject by org ID.

### What issues does this PR fix or reference?
@W-11831532@

QA suggestions:
Create a sandbox from the UI.  Auth to it from the CLI.  Check the auth file for the org and that a sandbox specific file was created in the global dir.  Attempting to delete the sandbox should work, since it can now be properly identified as a sandbox during authentication.
